src/promptflow-core/promptflow/_utils/flow_utils.py

```diff
@@ -184,6 +186,25 @@ def is_flex_flow(
     return isinstance(yaml_dict, dict) and "entry" in yaml_dict
 
 
+def resolve_flow_language(
+    *,
+    flow_path: Union[str, Path, PathLike, None] = None,
+    yaml_dict: Optional[dict] = None,
+    working_dir: Union[str, Path, PathLike, None] = None,
+) -> str:
+    """Get language of a flow. Will return 'python' for Prompty."""
+    if flow_path is None and yaml_dict is None:
+        raise UserErrorException("Either file_path or yaml_dict should be provided.")
+    if flow_path is not None and yaml_dict is not None:
+        raise UserErrorException("Only one of file_path and yaml_dict should be provided.")
+    if flow_path is not None:
+        flow_path, flow_file = resolve_flow_path(flow_path, base_path=working_dir, check_flow_exist=False)
+        file_path = flow_path / flow_file
+        if file_path.is_file() and file_path.suffix.lower() in (".yaml", ".yml"):
+            yaml_dict = load_yaml(file_path)
+    return yaml_dict.get(LANGUAGE_KEY, FlowLanguage.Python)
```

Comment:
when yaml_dict is none, it will raise exception

Reply:
If yaml_dict is None
1. if `flow_path` is also None, it will raise a UserException at the beginning
2. if `flow_path` is not None, yaml_dict will be replaced by what is loaded from `flow_path`
Or you have concern that there will be a flow_path pointed to a null yaml?

Reply:
got your point. add an extra error handling